### PR TITLE
feat(core): add shadow css variables

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

### CODE REVIEW

File: projects/core/components/theme-night/theme-night.style.less [hunk 1]
```diff
@@ -43,6 +43,14 @@
...
     --tui-elevation-01: #222;
     --tui-elevation-02: #222;
+    --tui-shadow-basic: 0 0.25rem 1.5rem rgba(0, 0, 0, 0.12);
+    --tui-shadow-basic-hover: 0 0.75rem 2.25rem rgba(0, 0, 0, 0.2);
+    --tui-shadow-dropdown: 0 1.5rem 1rem rgba(0, 0, 0, 0.03), 0 0.75rem 0.75rem rgba(0, 0, 0, 0.04),
+        0 0.25rem 0.375rem rgba(0, 0, 0, 0.05);
+    --tui-shadow-modal: 0 1.125rem 1.875rem rgba(0, 0, 0, 0.48);
+    --tui-shadow-sidebar: 0.25rem 0 1.5rem rgba(0, 0, 0, 0.12);
+    --tui-shadow-navigation: 0 0.125rem 1rem rgba(0, 0, 0, 0.08);
+    --tui-shadow-mobile-modal: 0 -1rem 1.75rem rgba(0, 0, 0, 0.24);
```
Comment (waterplea) [inline projects/core/components/theme-night/theme-night.style.less:53]: Maybe `--tui-shadow-sheet`?

File: projects/core/styles/mixins/mixins.less [hunk 2]
```diff
@@ -159,40 +159,37 @@
 .shadow(@mode: 1) {
```
Comment (waterplea) [inline projects/core/styles/mixins/mixins.less:159]: Let's deprecate the mixin

File: projects/core/components/theme-night/theme-night.style.less [hunk 3]
```diff
@@ -43,6 +43,14 @@
     --tui-autofill: var(--tui-autofill-night);
     --tui-elevation-01: #222;
     --tui-elevation-02: #222;
+    --tui-shadow-basic: 0 0.25rem 1.5rem rgba(0, 0, 0, 0.12);
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/core/components/theme-night/theme-night.style.less]: fixed

File: projects/core/components/theme-night/theme-night.style.less (hunk 1 above)
Comment (splincode) [reply projects/core/components/theme-night/theme-night.style.less]: fixed

File: projects/core/styles/mixins/mixins.less (hunk 2 above)
Comment (splincode) [reply projects/core/styles/mixins/mixins.less:160]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**